docs/data/routes/docs/nextjs/creating-components/new-component/en.md

```diff
@@ -0,0 +1,221 @@
+---
+name: new-component
+routeTemplate: ./data/component-templates/article.yml
+title: Walkthrough: Creating a new component
+---
+# Walkthrough: Creating a New Component
+
+This walkthrough demonstrates how to create a new component based on the chosen development workflow: 
+
+* [Using the Sitecore-first development workflow](#using-the-sitecore-first-development-workflow).
+* [Using the code-first development workflow](#using-the-code-first-development-workflow).
+
+## Using the Sitecore-first development workflow
+
+This walkthrough describes how to:
+
+- [Create the JSON rendering in Sitecore](#create-a-json-rendering-in-sitecore)
+- [Create the component in the Next.js app](#create-the-component-in-the-nextjs-app)
+- [Edit in Experience Editor and publish](#edit-in-experience-editor-and-publish)
+
+### Create the JSON rendering in Sitecore
+
+To create the JSON rendering in Sitecore:
+
+1. In the Content Editor, create a new template called `DataSourceExample` under `/sitecore/templates/Project/myproject`.
+
+2. Create a new template section (the name is unimportant) and add the following fields:
+
+   - `Title`: Single-line Text
+   - `BodyText`: Rich Text
+   
+3. With the **Builder** tab open, in the **Builder Options** menu, click **Standard Values** and in the **Title** and **BodyText** fields enter default values.
+
+4. Create a JSON rendering called `DataSourceExample` in `/sitecore/layout/Renderings/Project/myproject`. Enter the following values:
+
+   - **Datasource Location**: ./
+   - **Datasource Template**: Select `sitecore/Templates/Project/myproject/DataSourceExample`
+
+5. Add the `DataSourceExample` rendering to the **Allowed Controls** in the `/sitecore/layout/Placeholder Settings/Project/myproject/Main` placeholder and click **Save**.
+
+6. Open `/sitecore/content/MyProject/Home` in the Experience Editor and add your new rendering, including creating a data source item for it.
+
+   > Your rendering host page outputs `Unknown component` or a similar message because you have not yet mapped a component to this JSON rendering in your rendering host.
+
+7. Publish all your item changes:
+
+   - In Content Editor, on the **Publish** ribbon, click the small black arrow next to the **Publish** icon and click **Publish site**.
+   - In the **Publish Site** window, to publish your items from the Master database to the Web database, select the **Smart publish** radio button and click **Publish**.
+
+8. You can now test the output of your new rendering:
+
+   - In your rendering host, open the site home page.
+
+   - Type the following into a PowerShell terminal to get the rendering host logs: `docker-compose logs -f rendering`.
```

Comment:
Ahh actually this won't be the case for Next.js, you'd need to open the layout service directly. I will validate steps and add to PR.

Reply:
Thank you.